File: src/main/java/com/awsblog/queueing/model/PeekResult.java
package com.awsblog.queueing.model;

import com.awsblog.queueing.appdata.Assignment;
import com.awsblog.queueing.utils.Utils;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * Result for the peek() API call
 * 
 * @author zorani
 *
 *
 * Need to update this to reflect returning schedules
 */
@JsonInclude(Include.NON_NULL)
public class PeekResult extends ReturnResult {

	/**
	 * Default empty c-tor
	 */
	public PeekResult() {
		
		super();
	}

	/**
	 * C-tor
	 */
	public PeekResult(String id) {
		
		super(id);
	}
	
	/**
	 * @return the peekedShipmentObject
	 */
	public Assignment getPeekedAssignmentObject() {
		return peekedAssignmentObject;
	}

	/**
	 * @param peekedAssignmentObject the peekedShipmentObject to set
	 */
	public void setPeekedAssignmentObject(Assignment peekedAssignmentObject) {
		this.peekedAssignmentObject = peekedAssignmentObject;
	}	
	
	/**
	 * @return the timestampMillisUTC
	 */
	public long getTimestampMillisUTC() {
		return timestampMillisUTC;
	}

	/**
	 * @param timestampMillisUTC the timestampMillisUTC to set
	 */
	public void setTimestampMillisUTC(long timestampMillisUTC) {
		this.timestampMillisUTC = timestampMillisUTC;
	}

	/**
	 * Get the peeked shipment id
	 * 
	 * @return
	 */
	@JsonProperty("peeked_id")
	public String getPeekedAssignmentId() {
		
		if (Utils.checkIfNotNullObject(this.peekedAssignmentObject))
			return this.peekedAssignmentObject.getId();
		
		return "NOT FOUND";
	}

	@JsonProperty("peeked_schedule")
	public String getPeekedAssignmentSchedule(){
		if (Utils.checkIfNotNullObject(this.peekedAssignmentObject))
			return this.getPeekedAssignmentObject().getSchedule();

		return "NOT FOUND";
	}

	// ---------------- fields

	@JsonProperty("timestamp_milliseconds_utc")
	private long timestampMillisUTC = 0L;
	
	@JsonIgnore
	private Assignment peekedAssignmentObject = null;
	
} // end PeekResult

File: src/main/kotlin/com/awsblog/queueing/model/PeekResult.kt
package com.awsblog.queueing.model

import com.awsblog.queueing.appdata.Assignment
import com.awsblog.queueing.utils.Utils
import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonProperty

/**
 * Result for the peek() API call
 *
 * @author zorani
 *
 *
 * Need to update this to reflect returning schedules
 */
@JsonInclude(JsonInclude.Include.NON_NULL)
class PeekResult : ReturnResult {
    /**
     * Default empty c-tor
     */
    constructor() : super()

    /**
     * C-tor
     */
    constructor(id: String?) : super(id)

    @get:JsonProperty("peeked_id")
    val peekedAssignmentId: String?
        /**
         * Get the peeked shipment id
         *
         * @return
         */
        get() = if (Utils.checkIfNotNullObject(peekedAssignmentObject)) peekedAssignmentObject?.id else "NOT FOUND"

    @get:JsonProperty("peeked_schedule")
    val peekedAssignmentSchedule: String?
        get() = if (Utils.checkIfNotNullObject(peekedAssignmentObject)) peekedAssignmentObject!!.getSchedule() else "NOT FOUND"
    /**
     * @return the timestampMillisUTC
     */
    /**
     * @param timestampMillisUTC the timestampMillisUTC to set
     */
    // ---------------- fields
    @JsonProperty("timestamp_milliseconds_utc")
    var timestampMillisUTC = 0L
    /**
     * @return the peekedShipmentObject
     */
    /**
     * @param peekedAssignmentObject the peekedShipmentObject to set
     */
    @JsonIgnore
    var peekedAssignmentObject: Assignment? = null
} // end PeekResult
